kubesphere-sigs/ks-devops-helm-chart
# Move migrate job into init containers of devops apiserver

### What dose this PR?

Move migrate job into init containers of devops apiserver. `migrate-config` init container must be before `init-config`, because `migrate-config` will migrate password from ConfigMap `kubesphere-config` into `devops-config`, which password might be invalid.

### Why do we need it?

We cannot guarantee the execution order of Job and apiserver, so apiserver may not get the desired configuration normally.

### Steps to test

1. Create Kubernetes with KubeSphere
1. Add `sonarQube` configuration snippet into ConfigMap `kubesphere-config`

    ```yaml
        sonarQube:
            host: http://test.com
            token: test-token
    ```
2. Use helm to install ks-devops

    ```bash
    helm install ks-devops ks-devops/ks-devops -n kubesphere-devops-system --create-namespace \
		     --set image.pullPolicy=Always --set jenkins.ksAuth.enabled=true \
                     --wait
    ```

3. See the ConfigMap `devops-config` and logs of Pod `ks-devops-apiserver`

```diff
- I1201 15:13:13.033828       1 sonarqube.go:39] Sonarqube integration is disabled
- I1201 15:13:13.036337       1 sonarqube.go:39] Sonarqube integration is disabled
```

/kind enhancement
/cc @kubesphere-sigs/sig-devops
